### PR TITLE
Fixed issues  for device detail page

### PR DESCRIPTION
**Pull request by ajaypratap003**
### Type of change

- Bugfix
- Enhancement / new feature
- Refactoring

### Description
Fixed issues  for device detail page

### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: console/console-init/ui/src/modules/iot-device/containers/AddGatewaysContainer/AddGatewaysContainer.tsx [hunk 1]
```diff
@@ -23,42 +23,44 @@ export const AddGatewaysContainer: React.FC<{
...
      * TODO: implement save query
      */
     onCancel();
   };
 
+  const shouldDisabledSaveBUtton = () => {
+    if (gatewayGroups?.length > 0 || gatewayDevices?.length > 0) {
+      return false;
+    }
+    return true;
```
Comment (drcoolsanjeev) [inline console/console-init/ui/src/modules/iot-device/containers/AddGatewaysContainer/AddGatewaysContainer.tsx:37]: we can simply return `!(gateway?.length>0 || gatewayDevices?.length>0)`

File: console/console-init/ui/src/utils/utils.ts [hunk 2]
```diff
@@ -482,6 +484,15 @@ const getTitleForFailedQuery = (action: string) => {
   return "Server error";
 };
 
+const convertStringToBoolean = (value: string) => {
+  if (value === "true") {
```
Comment (drcoolsanjeev) [inline console/console-init/ui/src/utils/utils.ts:488]: we can simply return `value==="true"`

**Review by drcoolsanjeev — APPROVED** [review]
LGTM, added few suggestions

File: console/console-init/ui/src/modules/iot-device/containers/AddCredentialsContainer/AddCredentialsContainer.tsx [hunk 3]
```diff
@@ -50,6 +50,17 @@ export const AddCredentialsContainer: React.FC<{
     await setCredentialQueryVariable(variable);
   };
 
+  const shouldDisabledSaveBUtton = () => {
```
Comment (anukritijha) [inline console/console-init/ui/src/modules/iot-device/containers/AddCredentialsContainer/AddCredentialsContainer.tsx:53]: U of Button is capitalized.

Comment (anukritijha) [reply console/console-init/ui/src/modules/iot-device/containers/AddCredentialsContainer/AddCredentialsContainer.tsx:53]: I'd suggest renaming this to 'isSaveEnabled'. This change should reflect in all similar methods used elsewhere.

File: console/console-init/ui/src/modules/iot-device/containers/EditDeviceInJsonContainer/EditDeviceInJsonContainer.tsx [hunk 4]
```diff
@@ -39,6 +39,7 @@ export const EditDeviceInJsonContainer: React.FC<{
   const { projectname, deviceid, namespace } = useParams();
   const [deviceJson, setDeviceJson] = useState<any>();
   const [hasError, setHasError] = useState<boolean>(false);
+  const [isDisabledSaveButton, setDisabledSaveButton] = useState<boolean>(true);
```
Comment (anukritijha) [inline console/console-init/ui/src/modules/iot-device/containers/EditDeviceInJsonContainer/EditDeviceInJsonContainer.tsx:42]: Please rename to isSaveDisabled.

**Review by anukritijha — APPROVED** [review]
Added a couple of minor suggestions.

**Review by rkpattnaik780 — COMMENTED** [review]
Looks good, just a minor typo.

**Review by suyash-naithani — APPROVED** [review]

**Pull request merged**